docs/madr/decisions/012-health-check-policy.md

```diff
@@ -0,0 +1,180 @@
+
+##### Option 1 - allow everything
+
+###### Pros
+
+- Flexibility
+- Can avoid DOS-ing by using reachable services
+
+###### Cons
+
+- Users might accidentally DOS their infra by making Everything HC service X
+- Without reachable services using `targetRef=Mesh` and `to.targetRef=Mesh` does not make sense
+- Cases that might produce a lot of traffic need to be documented
+
+##### Option 2 - disallow Mesh-Mesh
+
+###### Pros
+
+- Flexibility (without 1 case)
+- Can avoid DOS-ing by using reachable services
+
+###### Cons
+
+- Users might accidentally DOS their infra by making Everything HC service X
+- Cases that might produce a lot of traffic need to be documented
+
+##### Option 3 - allow only MeshService
+
+###### Pros
+
+- Harder to accidentally DOS infrastructure by running too aggressive health checks
+
+###### Cons
+
+- Inflexibility
+
+#### Decision drivers
+
+As we don't pose any restrictions on HC right now I would go with **option 1** and an entry in docs about possible problems.
+
+### GRPC
+
+GRPC support seems pretty straight forward,
+just need to use a different health checking type - [GrpcHealthCheck](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/health_check.proto.html?highlight=grpc_health_check#envoy-v3-api-field-config-core-v3-healthcheck-grpc-health-check)
+
+### Configuration options
+
+There are some features that we currently do not expose (like pass through mode / caching, redis / thrift protocol),
+but currently this MADR does not aim to change that, if you think something critical is missing from this functionality,
+let us know in the PR.
+
+The question is - do we need `service_name`, `authority` and `initial_metadata` gRPC options in the first iteration?
+
+```yaml
+default:
+  interval: 10s
+  timeout: 2s
+  unhealthyThreshold: 3
+  healthyThreshold: 1
+  initialJitter: 5s # optional
+  intervalJitter: 6s # optional
+  intervalJitterPercent: 10 # optional
+  healthyPanicThreshold: 60 # optional, by default 50
+  failTrafficOnPanic: true # optional, by default false
+  noTrafficInterval: 10s # optional, by default 60s
+  eventLogPath: "/tmp/health-check.log" # optional
+  alwaysLogHealthCheckFailures: true # optional, by default false
+  reuseConnection: false # optional, by default true
+  tcp: # only one of tcp http or grpc can be enabled
+    enabled: true # new, default false, can be disabled for override
+    send: Zm9v # optional, empty payloads imply a connect-only health check
+    receive: # required if send specified
+    - YmFy
+    - YmF6
+  http:
+    enabled: true # new, default false, can be disabled for override
+    path: /health
+    requestHeadersToAdd: # optional, empty by default
+    - append: false
+      header:
+        key: Content-Type
+        value: application/json
+    - header:
+        key: Accept
+        value: application/json
+    expectedStatuses: [200, 201] # optional, by default [200]
```

Comment:
Should we add by default the right headers for things to work well with fault injection for example (the problem you had when working with demo). If yes we should explain it here in this MADR.

Reply:
I thought we decided to change how FaultInjection will work https://github.com/kumahq/kuma/issues/5212#issuecomment-1297262410 - maybe @jakubdyszkiewicz can confirm. If that can't be done then I'm happy to tackle this here.

Reply:
Ha right so FaultInjection will rely on: `X-Forwarded-Client-Cert` that's it? If yes then you are 100% right this doesn't need ot change

Reply:
yes